src/devices/Pcf8563/Pcf8563.cs

```diff
@@ -0,0 +1,118 @@
+        /// DS3231 DateTime
+        /// </summary>
+        public DateTime DateTime { get => ReadTime(); set => SetTime(value); }
+
+        /// <summary>
+        /// Creates a new instance of the PCF8563
+        /// </summary>
+        /// <param name="sensor">I2C Device, like UnixI2cDevice or Windows10I2cDevice</param>
+        public Pcf8563(I2cDevice sensor)
+        {
+            _sensor = sensor;
+
+            // Set "Normal Mode"
+            Span<byte> ctrl1Config = stackalloc byte[] { (byte)Register.PCF_CTRL1_ADDR, 0x00 };
+            _sensor.Write(ctrl1Config);
+        }
+
+        /// <summary>
+        /// Set PCF8563 Time
+        /// </summary>
+        /// <param name="time">Time</param>
+        private void SetTime(DateTime time)
+        {
+            Span<byte> writeBuffer = stackalloc byte[8];
+
+            writeBuffer[0] = (byte)Register.PCF_SEC_ADDR;
+            // Set bit8 as 0 to guarantee clock integrity
+            writeBuffer[1] = (byte)(Int2Bcd(time.Second) & 0b_0111_1111);
+            writeBuffer[2] = Int2Bcd(time.Minute);
+            writeBuffer[3] = Int2Bcd(time.Hour);
+            writeBuffer[4] = Int2Bcd(time.Day);
+            writeBuffer[5] = Int2Bcd((int)time.DayOfWeek);
+            if (time.Year >= 2000)
+            {
+                writeBuffer[6] = (byte)(Int2Bcd(time.Month) | 0b_1000_0000);
+                writeBuffer[7] = Int2Bcd(time.Year - 2000);
+            }
+            else
+            {
+                writeBuffer[6] = Int2Bcd(time.Month);
+                writeBuffer[7] = Int2Bcd(time.Year - 1900);
+            }
+
+            _sensor.Write(writeBuffer);
+        }
+
+        /// <summary>
+        /// Read Time from PCF8563
+        /// </summary>
+        /// <returns>PCF8563 Time</returns>
+        private DateTime ReadTime()
+        {
+            // Sec, Min, Hour, Date, Day, Month & Century, Year
+            Span<byte> readBuffer = stackalloc byte[7];
+
+            _sensor.WriteByte((byte)Register.PCF_SEC_ADDR);
+            _sensor.Read(readBuffer);
+
+            return new DateTime(readBuffer[5] >> 7 == 1 ? 2000 + Bcd2Int(readBuffer[6]) : 1900 + Bcd2Int(readBuffer[6]),
+                                Bcd2Int((byte)(readBuffer[5] & 0b_0001_1111)),
+                                Bcd2Int((byte)(readBuffer[3] & 0b_0011_1111)),
+                                Bcd2Int((byte)(readBuffer[2] & 0b_0011_1111)),
+                                Bcd2Int((byte)(readBuffer[1] & 0b_0111_1111)),
+                                Bcd2Int((byte)(readBuffer[0] & 0b_0111_1111)));
+        }
+
+        /// <summary>
+        /// Cleanup
+        /// </summary>
+        public void Dispose()
+        {
+            _sensor?.Dispose();
+            _sensor = null;
+        }
+
+        /// <summary>
+        /// BCD To Int
+        /// </summary>
+        /// <param name="bcd">BCD Code</param>
+        /// <returns>int</returns>
+        private int Bcd2Int(byte bcd)
```

Comment:
do we have this logic elsewhere? might be worth moving this to `common` directory

Reply:
I was thinking the same as it seems common in RTCs.

Reply:
optionally join all RTC into one if they are similar enough